water/include/water/Dialect/Wave/IR/WaveOps.td

```diff
@@ -215,6 +215,25 @@ def AllocateOp : WaveOp<"allocate"> {
   let hasVerifier = 1;
 }
 
+def ExtractSliceOp : Op<WaveDialect, "extract_slice"> {
+  let summary = "Takes an n-D vector, k-D `offsets` wave expression attribute,"
+    "a k-sized `sizes` wave expression attribute, a k-sized `strides` wave expression"
+    " attribute and extracts the n-D subvector at the proper offset.";
```

Comment:
You probably want to document (1) that `k <= n` and (2) which dimensions are being sliced with `k < n`.